clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2119,86 +2083,30 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
     return true;
   }
 
-  // FIXME: Refactor accessor handling. There is a discrepancy in how
-  // inherited accessors are handled.
-  bool handleSyclAccessorType(const CXXRecordDecl *, const CXXBaseSpecifier &BS,
-                              QualType FieldTy) final {
+  bool handleSyclSpecialType(const CXXRecordDecl *, const CXXBaseSpecifier &BS,
+                             QualType FieldTy) final {
     const auto *RecordDecl = FieldTy->getAsCXXRecordDecl();
-    assert(RecordDecl && "The accessor/sampler must be a RecordDecl");
-    llvm::StringLiteral MethodName = KernelDecl->hasAttr<SYCLSimdAttr>()
-                                         ? InitESIMDMethodName
-                                         : InitMethodName;
-    CXXMethodDecl *InitMethod = getMethodByName(RecordDecl, MethodName);
-    assert(InitMethod && "The accessor/sampler must have the __init method");
-
-    // Get access mode of accessor.
-    const auto *AccessorSpecializationDecl =
-        cast<ClassTemplateSpecializationDecl>(RecordDecl);
-    const TemplateArgument &AccessModeArg =
-        AccessorSpecializationDecl->getTemplateArgs().get(2);
+    assert(RecordDecl && "The type must be a RecordDecl");
+    CXXMethodDecl *InitMethod =
```

Comment:
I see you replaced calls to `__init_esimd `with `__init`. Won't this break ESIMD handling since we don't have the  corresponing header/BE changes? @kbobrovs

Reply:
Yes this is breaking the ESIMD tests. Not sure what should be done here.

Reply:
Me neither.  I think we have 2 options - 

1. Whatever FE+BE+header changes required to replace `__init_esimd` with `__init`, should be done before this patch.
2.  We revert this code to what you had earlier, and we crash on streams/samplers.  In this option we introduce new crashes in stream/sampler base class handling (field handler flow apparently always crashed) but we never supported base class handling properly IIRC. We just silently ignored it.

Don't know if there is some other option I am not thinking of. @bader @AaronBallman thoughts?

Reply:
I would opt for option 2, because of the new pressing request from PSG to use this attribute. And open a jira for both new and old crashes.

Reply:
I'm not keen on introducing new crashes that we file JIRAs for, so I lean towards Option 1. Assuming we do not have to worry about anyone mixing and matching the compiler and the headers, I think we can modify those headers as part of this patchset (so everything lands at the same time). Alternatively, we could use macros to support both the old way and the new way in the headers and land those changes separate from this patch.